happi/loader.py

```diff
@@ -201,6 +205,12 @@ def create_arg(arg):
         except Exception:
             logger.warning("Unable to attach metadata dictionary to device")
 
+        if run_post_attach_hook and hasattr(obj, 'post_happi_md'):
+            try:
+                obj.post_happi_md()
```

Comment:
I think no args is the correct decision here
We could have made `md` the arg, but as it's not something we're introducing in this step I think it's correct to let the user find self.md if they want to access it in the hook.

One design question: should this still run if the md setattr fails? There's a check for this above.
I guess the state where someone has made a device that can't get an md setattr'd but decided to include a post-hook is pretty low

Reply:
Makes sense to me, let's keep it as-is